File: src/main/java/co/gatedaccess/web/data/repo/SecurityGuardOtpRepo.java
package co.gatedaccess.web.data.repo;

import co.gatedaccess.web.data.model.SecurityGuardOtp;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface SecurityGuardOtpRepo extends MongoRepository<SecurityGuardOtp, String> {
    SecurityGuardOtp findByCode(String code);
    SecurityGuardOtp findByCommunityId(String communityId);
    void deleteByCommunityId(String communityId);
}

File: src/main/kotlin/co/gatedaccess/web/data/repo/SecurityGuardOtpRepo.kt
package co.gatedaccess.web.data.repo

import co.gatedaccess.web.data.model.SecurityGuardOtp
import org.springframework.data.mongodb.repository.MongoRepository

interface SecurityGuardOtpRepo : MongoRepository<SecurityGuardOtp?, String?> {
    fun findByCode(code: String?): SecurityGuardOtp?
    fun findByCommunityId(communityId: String?): SecurityGuardOtp?
    fun deleteByCommunityId(communityId: String?)
}
